### ISSUE TITLE
Protection error in do_wcentre

### ISSUE THREAD

**Issue by jarioksa** [vegandevs/vegan#653]
CRAN checks for `vegan_2.6-6` contains "additional issue" `rchk`:

```
Package vegan version 2.6-6
Package built using 86566/R 4.5.0; x86_64-pc-linux-gnu; 2024-05-20 16:03:08 UTC; unix   
Checked with rchk version fdc068715daa3a256062cc20e0d4a5157dacc9a4 LLVM version 14.0.6
More information at https://github.com/kalibera/cran-checks/blob/master/rchk/PROTECT.md
For rchk in docker image see https://github.com/kalibera/rchk/blob/master/doc/DOCKER.md

Function do_wcentre
  [UP] unprotected variable x while calling allocating function Rf_coerceVector vegan/src/goffactor.c:83
  [UP] unprotected variable x while calling allocating function Rf_duplicate vegan/src/goffactor.c:84
  [UP] unprotected variable x while calling allocating function Rf_allocMatrix vegan/src/goffactor.c:85
```

Here `[UP]` means unprotected SEXP pointer.

Protection error means that garbage collection can destroy the pointer before it is used, and this can segfault R (not only vegan). I may have seen this: reverse dependencies test sometimes fails with `vectorfit` in testing `flowCHIC`.

This also means that I need learn new skills and read a bit.

**Comment by jarioksa** [comment]
I think this is fixed in commit 33b4c3379ee2bfb0bbdf06443d64e0cb487b052f (cherry-picked to 2.7-0 or master) and a bug-fix version 2.6-6.1 was submitted to CRAN. I checked the fix with Tomas Kalibera's Docker image (`kalibera/rchk`), but I'm waiting for CRAN team's confirmation.

**Comment by jarioksa** [comment]
The fix was blessed by the CRAN team, and `vegan_2.6-6.1` is on the CRAN.
